pkg/kubecost/asset.go

```diff
@@ -1936,10 +1938,53 @@ func (n *Node) Add(a Asset) Asset {
 
 func (n *Node) add(that *Node) {
 	if n == nil {
-		n = that
 		return
 	}
 
+	// If we detect that we are accumulating a unique node, then we can make
+	// stronger assumptions later on that may allow us to correct for inaccuracies
+	// in accumulated data (e.g. cascading data pipelines) by interpolating
+	// correct data where it was briefly lost.
+	// For more detail, see ELV-102 https://kubecost.atlassian.net/browse/ELV-102
+	accumulatingUniqueNode := false
+	var thisCPUCores, thisRAMBytes float64
+	var thatCPUCores, thatRAMBytes float64
+	if n.Properties.ProviderID != "" && n.Properties.ProviderID == that.Properties.ProviderID {
+		accumulatingUniqueNode = true
+
+		// CPU cores and RAM bytes for each should match. If not, log a debug
+		// statement. If these values match, but the node resulting from the
+		// addition does not match, then we are missing data and can fix that
+		// by interpolation.
+		//
+		// e.g. if we are accumulating two hours of data, for the same node,
+		// which has 2 CPU cores and 8 GiB RAM, but Kubecost was down for the
+		// final 20m and the first 30m of the back-to-back hours, then without
+		// correction we would yield:
+		//  - (end-start) = 2 hours
+		//  - CPU = 1.3333 + 1 = 2.3333 core-hours == 1.16667 cores (58%)
+		//  - RAM = 5.3333 + 4 = 9.3333 gib-hours  == 4.66667 GiB (58%)
+		//
+		// We will also end up having only 58% of the actual RAM and CPU costs.
+		// No good! But we can detect this (by comparing capacities for these,
+		// before and after, which should never change) and scaling costs and
+		// capacities up to 100%.
+
+		thisCPUCores = n.CPUCores()
```

Comment:
I see what you are doing here. At this point in the code, you can trust that thisCPUCores is accurate since the cpu core hours should line up with the node start end, and we are at 'maximum granularity', before expanding our windows and such and really starting the add operation. This lets you accurately describe the scale factor later on. It helped to think of thisCPUCores as a variable like cpuCoresPreAdd or something. 

It is after you adjust the windows and start/end times that the mismatching starts to happen since the existing logic made assumptions about start and end times which you are now interpolating here. 

reading over your comment 
```
                //If these values match, but the node resulting from the
		// addition does not match, then we are missing data and can fix that
		// by interpolation.
```

I don't see logic where you are enforcing the if condition that snippet started with - if the values don't match, you log an error but still continue the operation. Maybe comment needs updating? What action would you take if these don't match